packages/server/src/api/notification/notification-alert.controller.ts

```diff
@@ -0,0 +1,314 @@
+        .andWhere('observation.timestamp BETWEEN :minTime AND :maxTime', {
+          minTime: moment().subtract(minutes, 'minutes'),
+          maxTime: moment(),
+        })
+        .andWhere(
+          `((observation.custom_columns->>'Score')::int >= 20 
+          OR (observation.custom_columns->>'TemperatureFahrenheit')::int >= 102
+          OR observation.custom_columns->>'Category' In (:...keywords))`,
+          {
+            keywords: ['medium', 'high', 'very High'],
+          },
+        )
+        .groupBy('observation.unit')
+        .getRawMany();
+      rows.forEach((row: any) => {
+        // only add counts for units where the threshold is exceeded
+        if (row.count >= setting.threshold) {
+          alertData[row.unit] = row.count;
+        }
+      });
+
+      // only return records with actual alert data
+      if (!_.isEmpty(alertData)) {
+        output.push({
+          userSetting: setting,
+          alertData,
+        });
+      }
+    }
+    res.json(output);
+  }
+
+  // return the number of symptoms per unit that are trending upwards or downwards over last 3 days
+  async getNumTrendingSymptomsByUnit(req: ApiRequest<OrgSettingParams>, res: Response) {
+    const output: OutputRecord[] = [];
+
+    // first get all the users from the org who are registered for the given alert
+    const userSettings = (await getUsersRegisteredForNotification(req.appOrg!.id, 'symptomTrendingAlert'))
+      .filter((setting: UserNotificationSetting) => {
+        return isEligibleForNotification(setting);
+      });
+    // each property of this type is a symptom name with an array of counts indexed by day
+    type SymptomCountsByDay = { [key: string]: number[] };
+    // each property of this type is a unit name with the symptom counts by day
+    type UnitSymptomCounts = { [key: string]: SymptomCountsByDay };
+    const unitSymptomCountsByDay: UnitSymptomCounts = {};
+    const units = await Unit.find({
+      select: ['name'],
+      relations: ['org'],
+      where: { org: { id: req.appOrg!.id } },
+    });
+
+    // initialize the return object with all the units first
+    units.forEach((unit: Unit) => {
+      unitSymptomCountsByDay[unit.name] = {};
+    });
+
+    // make 3 queries for the past 3 days, iterating from past to recent
+    const lookBackDays = 3;
+    for (let lookBackIndex = lookBackDays; lookBackIndex > 0; lookBackIndex--) {
+      const rows = await Observation.createQueryBuilder('observation')
+        .select(`observation.custom_columns->>'Symptoms' as symptoms, observation.unit`)
+        .where('observation.report_schema_org = :org', { org: req.appOrg!.id })
+        .andWhere(`observation.report_schema_id = 'es6ddssymptomobs'`)
+        .andWhere('observation.timestamp BETWEEN :minTime AND :maxTime', {
+          minTime: moment().subtract(lookBackIndex, 'days'),
+          maxTime: moment().subtract(lookBackIndex - 1, 'days'),
+        })
+        .orderBy('observation.unit')
```

Comment:
Same as above, select from roster history and filter by user permissions for the final alert.